src/test/kotlin/com/pubnub/contract/crypto/CryptoModuleSteps.kt

```diff
@@ -0,0 +1,206 @@
+                cryptoModuleState.decryptedData = convertInputStreamToByteArray(decryptedStream)
+            }
+            else -> throw PubNubException("Invalid decryptionType type. Should be binary or stream")
+        }
+    }
+
+    private fun createCryptor(cryptorType: String, cipherKey: String, useRandomIv: Boolean): Cryptor {
+        return when (cryptorType) {
+            LEGACY_NEW -> {
+                LegacyCryptor(cipherKey, useRandomIv)
+            }
+            AES_CBC -> {
+                AesCbcCryptor(cipherKey)
+            }
+            else -> {
+                throw PubNubException("Invalid cryptor type")
+            }
+        }
+    }
+
+    @Then("I receive {string}")
+    fun I_receive_outcome(outcome: String) {
+        when (outcome) {
+            "unknown cryptor error" -> {
+                assertTrue(cryptoModuleState.decryptionError == PubNubError.UNKNOWN_CRYPTOR || cryptoModuleState.decryptionError == PubNubError.CRYPTOR_HEADER_VERSION_UNKNOWN)
+            }
+            "decryption error" -> assertEquals(
+                PubNubError.CRYPTOR_DATA_HEADER_SIZE_TO_SMALL,
+                cryptoModuleState.decryptionError
+            )
+            "success" -> assertNull(cryptoModuleState.decryptionError)
+        }
+    }
+
+    @Then("Successfully decrypt an encrypted file with legacy code")
+    fun successfully_decrypt_an_encrypted_file_with_legacy_code() {
+        val encryptedData = cryptoModuleState.encryptedData
+        val encryptedDataAsStringBase64 = String(Base64.encode(encryptedData, Base64.NO_WRAP))
+        val randoIv: Boolean = cryptoModuleState.initializationVectorType == RANDOM_IV
+        val crypto = Crypto(cryptoModuleState.cryptorCipherKey, randoIv)
+        val decryptedDataAsString: String = crypto.decrypt(encryptedDataAsStringBase64)
+        assertEquals(String(cryptoModuleState.fileContent!!), decryptedDataAsString)
+    }
+
+    @Then("Decrypted file content equal to the {string} file content")
+    fun decrypted_file_content_equal_to_the_source_file_content(sourceFileName: String) {
+        val sourceFileContent = getFileContentAsByteArray(sourceFileName)
+        assertArrayEquals(sourceFileContent, cryptoModuleState.decryptedData)
+    }
+
+    private fun getFileContentAsByteArray(fileName: String): ByteArray {
+        val cucumberOptions = RunMainCucumberTest.getCucumberOptions()
+        val featureFileLocation = cucumberOptions?.features?.joinToString()
+        val featureFilePath = Paths.get(featureFileLocation!!)
+        val cryptoAssetsLocation = featureFilePath.resolve("encryption/assets")
+        val fileToRead: Path = cryptoAssetsLocation.resolve(fileName)
+        return readFileAsByteArray(fileToRead.toString())
+    }
+
+    private fun readFileAsByteArray(filePath: String): ByteArray {
+        return File(filePath).inputStream().readBytes()
+    }
+
+    private fun convertInputStreamToByteArray(inputStream: InputStream): ByteArray {
```

Comment:
It's not needed. Check https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/java.io.-input-stream/read-bytes.html

Reply:
Thank's. Good to know.